osu.Game.Tests/Visual/Navigation/TestSceneButtonSystemNavigation.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Testing;
+using osu.Game.Screens.Menu;
+using osu.Game.Screens.Select;
+using osuTK.Input;
+
+namespace osu.Game.Tests.Visual.Navigation
+{
+    public class TestSceneButtonSystemNavigation : OsuGameTestScene
```

Comment:
I'll note that `TestSceneScreenNavigation` already exists, is a `OsuGameTestScene`, and these test cases would fit there, so I'm not sure this needs to be a test scene on its own. Although that is a bit of a mixed bag of tests right now.